main.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ModuleLength
+
+# Maru's Enumerables
+module Enumerable
+  def my_each
+    if block_given?
+      size.times do |i|
+        is_a?(Range) ? yield(min + i) : yield(self[i])
+      end
+    else
+      self
```

Comment:
here `self` would be returning the array the method it's evaluating, for example:

```
irb(main):116:0> [1,2,3].my_each
=> [1, 2, 3]
```

but we want

```
irb(main):117:0> [1,2,3].each
=> #<Enumerator: [1, 2, 3]:each>
```

therefore, maybe we should transform self into a enumerator, something like `self.to_enum`  to get:

```
irb(main):118:0> [1,2,3].to_enum
=> #<Enumerator: [1, 2, 3]:each>
```